basic_hashtable/b_hashtables.c

```diff
@@ -104,6 +129,11 @@ void hash_table_remove(BasicHashTable *ht, char *key)
  ****/
 char *hash_table_retrieve(BasicHashTable *ht, char *key)
 {
+  int h = hash(key, ht->capacity);
+  if (ht->storage[h] != NULL)
+  {
+    return ht->storage[h]->value;
+  }
   return NULL;
```

Comment:
This is technically what the comment called for, but a message saying unable to retrieve is not a bad idea :)